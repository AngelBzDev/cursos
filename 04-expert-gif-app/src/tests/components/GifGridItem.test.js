import {shallow} from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

   const title = 'Los simpson';
   const url = 'https://unaimagen_equis';
   const wrapper = shallow(<GifGridItem title={title} url={url}/>)
   
   test('Debe de mostrar el componente correctamete', () => {
      expect(wrapper).toMatchSnapshot();
   })
   
   test('Debe de tener un parrafo con el titulo', () => {
      const p = wrapper.find('p');
      expect(p.text().trim()).toBe(title);
   })

   test('Debe de tener la imagen la url y alt de los props', () => {
      const img = wrapper.find('img');
      expect(img.props().src).toBe(url);
      expect(img.props().alt).toBe(title);
   })
   
   test('Debe de tener la clase card', () => {
      const div = wrapper.find('div');
      expect(div.props().className).toBe('card');
   })
   

})
