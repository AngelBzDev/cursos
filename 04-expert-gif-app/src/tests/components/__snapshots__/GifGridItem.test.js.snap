// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar el componente correctamete 1`] = `
<div
  className="card"
>
  <img
    alt="Los simpson"
    src="https://unaimagen_equis"
  />
  <p>
    Los simpson
  </p>
</div>
`;
